### PR TITLE
Email notifications

### PR DESCRIPTION
**Pull request by distalx**
This PR is a part of an issue #626 and follows #776 

Subscribers will be notified on following events
- new hangouts 
   when user creates a hangout in a study group.
- new member
   when new members joins the group.
- new discussion 
   when user creates a discussion in a study group.

User can set notification preference from their account settings.

A cron job riggers these notifications.
Right now cron triggers at every 40 minute but for production we might want to tune it up to 20!.

### CODE REVIEW

File: server/hangouts/methods.js [hunk 1]
```diff
@@ -88,6 +88,11 @@ Meteor.methods({
         id: group._id,
         title: group.title,
         slug: group.slug
+      },
+      email_notifications: {
+        initial: false,
+        reminder: false,
+        follow_up: false
```
Comment (lpatmo) [inline server/hangouts/methods.js]: Does this mean users of a group won't receive email alerts when a hangout in that group is created?

File: imports/libs/server/cron/initial_email_notifications.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+               }
+
+              CBMailer(mail_data, template_name, template_data);
+            })
+
+
+            // set intial flag for email_notifications true
+            listOfNewHangouts.forEach(function(hangout){
+              Hangouts.update({_id:hangout._id}, {$set:{
+                'email_notifications.initial': true
```
Comment (lpatmo) [inline imports/libs/server/cron/initial_email_notifications.js]: .... oh wait, it's set to true here!

**Comment by lpatmo** [comment]
Looks beautiful.

One bug I noticed: if you start a new discussion from the nav bar (instead of from inside a study group), the attempt fails because the discussion needs to be linked with a groupId. 

Maybe the form for a new discussion started on the nav header needs to be different -- or include the list of groups?

Or perhaps we can remove the "Start a new discussion" button for now, and tell users to start new discussions from inside their groups.

File: imports/libs/server/cb_mailer/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { Meteor } from 'meteor/meteor';
+import {SSR} from 'meteor/meteorhacks:ssr';
```
Comment (nalbina) [inline imports/libs/server/cb_mailer/index.js]: whitespace

File: imports/libs/server/cb_mailer/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    from: mail_data.from,
+    subject: mail_data.subject,
+    html: SSR.render(template_name, template_data)
+  }
+
+  try {
+    Email.send(mail);
+  } catch (e) {
+    console.log(e);
+
```
Comment (nalbina) [inline imports/libs/server/cb_mailer/index.js]: whitespace

File: imports/libs/server/cron/helpers.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+/**
+* get list of users who are member of a group
+* @function
+* @name getGroupMembers
+* @param { String } type - email notification type
+* @param { Array } subscribers - list of user(s)
+* @return { Array } members - list of group members
+*/
+function getGroupMembers(study_group_id) {
+  if (study_group_id === 'CB') {
```
Comment (nalbina) [inline imports/libs/server/cron/helpers.js]: ?

File: imports/libs/server/cron/initial_email_notifications.js [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  createdFromNow } from './helpers';
+
+
+/**
+* send email notification for new hangouts to eligible recipients
+* @function
+* @name initialEmailNotificationsForHangouts
+*/
+async function initialEmailNotificationsForHangouts() {
+   let hangouts = await getNewHangouts();
```
Comment (nalbina) [inline imports/libs/server/cron/initial_email_notifications.js]: maybe this is worth wrapping in try catch.

File: server/hangouts/methods.js (hunk 1 above)
Comment (distalx) [reply server/hangouts/methods.js]: No, These flags are there so cron job can keep track for which notification has been sent.
we makes it true once notification is sent. [here](https://github.com/codebuddies/codebuddies/pull/801/files/8d65e59ddf86bfbc9e7a295b7252a85827ef7ff9#diff-6c4f4a0e79a2047c67141624ce02c65aR73)

**Comment by distalx** [comment]
@lpatmo  Thank you for reviewing this.

>One bug I noticed: if you start a new discussion from the nav bar (instead of from inside a study group), the attempt fails because the discussion needs to be linked with a groupId.

In that case Discussion is linked to the `CB`(global) group.

>Maybe the form for a new discussion started on the nav header needs to be different -- or include the list of groups? Or perhaps we can remove the "Start a new discussion" button for now, and tell users to start new discussions from inside their groups.

The idea behind this behaviour is that user can start a discussion without being a part of any group. same as a hangout. and yes we should include the list of groups.

File: imports/libs/server/cron/initial_email_notifications.js (hunk 6 above)
Comment (distalx) [reply imports/libs/server/cron/initial_email_notifications.js]: `getNewHangouts` doesn't throw any err, So I'm not sure what purpose would it serve.

**Pull request merged**